sqldelight-idea-plugin/src/main/kotlin/com/squareup/sqldelight/intellij/inspections/UnusedQueryInspection.kt

```diff
@@ -0,0 +1,95 @@
+package com.squareup.sqldelight.intellij.inspections
+
+import com.alecstrong.sql.psi.core.psi.SqlTypes
+import com.intellij.codeInspection.LocalInspectionTool
+import com.intellij.codeInspection.LocalInspectionToolSession
+import com.intellij.codeInspection.LocalQuickFixOnPsiElement
+import com.intellij.codeInspection.ProblemHighlightType
+import com.intellij.codeInspection.ProblemsHolder
+import com.intellij.openapi.command.WriteCommandAction
+import com.intellij.openapi.editor.ReadOnlyModificationException
+import com.intellij.openapi.module.ModuleUtil
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiElementVisitor
+import com.intellij.psi.PsiFile
+import com.intellij.psi.SmartPointerManager
+import com.intellij.psi.search.FilenameIndex
+import com.intellij.psi.search.GlobalSearchScope
+import com.intellij.psi.search.searches.ReferencesSearch
+import com.intellij.psi.util.elementType
+import com.squareup.sqldelight.core.lang.SqlDelightFile
+import com.squareup.sqldelight.core.lang.psi.StmtIdentifierMixin
+import com.squareup.sqldelight.core.lang.queriesName
+import com.squareup.sqldelight.core.lang.util.findChildrenOfType
+import com.squareup.sqldelight.core.psi.SqlDelightStmtIdentifier
+import com.squareup.sqldelight.core.psi.SqlDelightVisitor
+import org.jetbrains.kotlin.asJava.toLightMethods
+import org.jetbrains.kotlin.psi.KtNamedFunction
+
+class UnusedQueryInspection : LocalInspectionTool() {
+
+  override fun buildVisitor(
+    holder: ProblemsHolder,
+    isOnTheFly: Boolean,
+    session: LocalInspectionToolSession
+  ): PsiElementVisitor = object : SqlDelightVisitor() {
+
+    override fun visitStmtIdentifier(o: SqlDelightStmtIdentifier) {
+      if (o !is StmtIdentifierMixin || o.identifier() == null) {
+        return
+      }
+      val module = ModuleUtil.findModuleForPsiElement(o) ?: return
+      val sqlDelightFile = o.containingFile as SqlDelightFile
+      val fileName = "${sqlDelightFile.virtualFile?.queriesName}.kt"
+      val generatedFile = FilenameIndex.getFilesByName(o.project, fileName, GlobalSearchScope.moduleScope(module))
+        .firstOrNull() ?: return
+      val generatedMethods = generatedFile.findChildrenOfType<KtNamedFunction>()
+        .filter { namedFunction -> namedFunction.name == o.identifier()?.text }
+      for (generatedMethod in generatedMethods) {
+        val lightMethods = generatedMethod.toLightMethods()
+        for (psiMethod in lightMethods) {
+          val reference = ReferencesSearch.search(psiMethod, psiMethod.useScope).findFirst()
+          if (reference != null) {
+            return
+          }
+        }
+      }
```

Comment:
find usages should just work for queries, did you try just doing `ReferencesSearch.search(o, GlobalSearchScope.allScope())`?

Reply:
I just tried this and it doesn't work